annotations/component-annotations/src/main/java/io/ap4k/component/handler/ComponentHandler.java

```diff
@@ -62,37 +56,38 @@ public boolean canHandle(Class<? extends Configuration> type) {
       type.equals(EditableCompositeConfig.class);
   }
 
-  private void addVisitors(CompositeConfig config) {
```

Comment:
Using the visitor approach  allows us to modify existing resources, rather than just replacing them.

I am not sure if this makes much sense for the Component but I also see no harm either.

Reply:
We went for the simpler code for now, will revert to using visitors if proven needed later on.